tests/views/test_collection_exercise.py

```diff
@@ -146,7 +146,35 @@ def test_no_upload_collection_instrument_when_bad_extension(self, mock_request):
 
         self.assertEqual(response.status_code, 200)
         self.assertNotIn("Collection instrument loaded".encode(), response.data)
-        self.assertIn("Error: wrong file type for Collection instrument".encode(), response.data)
+        self.assertIn("Error: wrong file type for collection instrument".encode(), response.data)
+
+    @requests_mock.mock()
+    def test_no_upload_collection_instrument_when_bad_form_type_format(self, mock_request):
+        post_data = {
+            'ciFile': (BytesIO(b'data'), '064_201803_xxxxx.xlsx'),
+            'load-ci': '',
+        }
+        mock_request.get(url_get_collection_exercise, json=collection_exercise_details)
+
+        response = self.app.post("/surveys/test/000000", data=post_data)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertNotIn("Collection instrument loaded".encode(), response.data)
+        self.assertIn("Error: invalid file name format for collection instrument".encode(), response.data)
+
+    @requests_mock.mock()
+    def test_no_upload_collection_instrument_bad_file_name_format(self, mock_request):
+        post_data = {
+            'ciFile': (BytesIO(b'data'), '064201803_xxxxx.xlsx'),
+            'load-ci': '',
+        }
+        mock_request.get(url_get_collection_exercise, json=collection_exercise_details)
+
+        response = self.app.post("/surveys/test/000000", data=post_data)
+
+        self.assertEqual(response.status_code, 200)
```

Comment:
we get a 200 response for a bad file format with no upload? This should be a 400 bad request response.

Reply:
I think it's 200 because it still renders the page successfully